File: tests/data/example_plugins/example_plugin.py
from __future__ import annotations


from rpp_plugin_types.rpp_common import MotionController2D
from rpp_plugin_types.rpp_common import DisturbanceGenerator2D


COMPONENTS = {
    "ctl_main": "rpp_common::MotionController2D",
    "ctl_disturbance": "rpp_common::DisturbanceGenerator2D",
}

class ComponentPluginPy(MotionController2D):
    def __init__(self, name: str):
        super().__init__(name)


    def step(self, dt: float, disturbance: DisturbanceGenerator2D) -> None:
        # Implement the control logic here
        pass

File: tests/data/example_plugins/example_plugin.cpp
#include <rpp_plugin_types/rpp_testing/MotionController2D.hpp>


std::map<std::string, std::string> COMPONENTS = {
    {"ctl1", "rpp_testing::MotionController2D"},
};

class ComponentPlugin : public rpp_testing::MotionController2D
{

public:
    ComponentPlugin() = default;

    virtual ~ComponentPlugin() = default;

    rpp_testing::MotionController2D::VectorPlanar step(rpp_testing::MotionController2D::Pose2D state, double dt) override
    {
        auto a = 5;
    }

};


